Sources/ClientRuntime/Identity/IdentityAPI/IdentityResolver.swift

```diff
@@ -0,0 +1,15 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+
+// Base protocol for all identity provider types
+public protocol IdentityResolver {
+    associatedtype IdObj: Identity
```

Comment:
nit:  Not sure the significance of the name `IdObj` for this associated type.

Better to name this `IdentityT` which matches SRA?

Reply:
It's short for "Identity Object" - but I suppose it would be good to include `T` at end to notate that it's a generic placeholder + to match SRA. Will make change and push it up.